tool/tsh/kube.go

```diff
@@ -913,20 +928,10 @@ func (c *kubeLoginCommand) run(cf *CLIConf) error {
 		return trace.NotFound("kubernetes cluster %q not found, check 'tsh kube ls' for a list of known clusters", c.kubeCluster)
 	}
 
-	// Try updating the active kubeconfig context.
-	if err := kubeconfig.SelectContext(currentTeleportCluster, c.kubeCluster); err != nil {
```

Comment:
Why is the SelectContext not being called anymore? Do we set active context elsewhere now?

Reply:
The `SelectContext` method was used to switch existing contexts in kubeconfig without changing anything else other than `current-context` field. 

If the context existed in kubeconfig then the method overrode `current-context`. If it didn't exist, the `updateKubeConfig` method was called to add the new context and select it as the `current-context`.

Since we now have the flags we can't just update only the `current-context` field otherwise it wouldn't be possible to remove the `--as` and `--as-groups`  values if the context already existed in the kubeconfig.

The following block will rewrite the kubeconfig section and change the active context without checking if the context exists or not.

```if err := updateKubeConfig(cf, tc, ""); err != nil {
		return trace.Wrap(err)
	}```